modules/@apostrophecms/doc/index.js

```diff
@@ -1752,14 +1751,10 @@ module.exports = {
               }
             } else if (field.type === 'object') {
               const value = doc[field.name];
-              if (value) {
-                handlers.object(field, value);
-                forSchema(field.schema, value);
-              }
+              handlers.object(field, value);
+              forSchema(field.schema, value);
             } else if (field.type === 'relationship') {
-              if (Array.isArray(doc[field.name])) {
-                handlers.relationship(field, doc);
-              }
+              handlers.relationship(field, doc, true);
```

Comment:
Since for relationships I need to pass inside the handler when I have no `doc[field.name]` in my document. I decided to move the logic that checks if the value exists inside the handlers of `prepareForStorage`.
This way I can reuse this method for `simulateRelationshipsFromStorage`.